staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -373,13 +373,8 @@ func dedupDeltas(deltas Deltas) Deltas {
 	a := &deltas[n-1]
 	b := &deltas[n-2]
 	if out := isDup(a, b); out != nil {
-		// `a` and `b` are duplicates. Only keep the one returned from isDup().
-		// TODO: This extra array allocation and copy seems unnecessary if
-		// all we do to dedup is compare the new delta with the last element
-		// in `items`, which could be done by mutating `items` directly.
-		// Might be worth profiling and investigating if it is safe to optimize.
-		d := append(Deltas{}, deltas[:n-2]...)
-		return append(d, *out)
+		deltas[n-2] = *out
```

Comment:
the signature of dedupDeltas does not make it obvious the input slice will be modified... that seems like it could be surprising or lead to subtle bugs

Reply:
I think the old code overwrote deltas[n-2] already, though.

OK fine I verified that append works like I thought: https://play.golang.org/p/9n1QmE8VxdT